src/components/data/data-utils.js

```diff
@@ -36,6 +36,25 @@ const errorTextStyle = { color: colors.danger(), fontWeight: 'bold', fontSize: 1
 
 export const parseGsUri = uri => _.drop(1, /gs:[/][/]([^/]+)[/](.+)/.exec(uri))
 
+export const getDownloadCommand = (fileName, gsUri, accessUrl) => {
```

Comment:
Since you seem to be accessing properties of `accessUrl` a lot, does it make sense to do some preemptive destructuring here and in `UriViewer`? Something like:

```js
const exampleFnDeclaration = (accessUrl: { url: accessUrl, headers: accessHeaders } = {}) => {}
```

(I added some renaming for clarity)

Reply:
Meybe? I feel like there was some comment about the number of variables to help the browser... I'll try to find the old comment that I may have misinterpreted.

Reply:
Found it- here's the previous comment about creating more variables: [derefed link](http://www.dereferer.org/?https%3A%2F%2Fgithub%2Ecom%2FDataBiosphere%2Fterra%2Dui%2Fpull%2F2233%23discussion%5Fr487900931).

So it had something to do with `state`, that may not be applicable here? TODO: At some point if I'm going to be ~tau~ tee or ~table~ pi shaped I should take some time and do a react course I guess...

Reply:
Ok, I think I see the difference. You suggestion is to add additional destructuring for only local function variables vs. additional destructuring into whatever React's `state` is... I think. I'm assuming `state` is lurking around the browser memory as the name implies.

Reply:
Right; state sticks around from render to render, whereas this is just sugar quick access to contents of things, and can be handled by the optimization when code gets minified automagically if appropriate.

Reply:
Gave passing by name plus destructuring a shot. Had to figure out how to get rid of `null` parameter values so they would be defaulted. Found something on StackOverflow so let me know if there's something more modern?